src/cpp/include/openvino/genai/generation_config.hpp

```diff
@@ -0,0 +1,107 @@
+ * @param length_penalty exponential penalty to the length that is used with beam-based generation. It is applied as an exponent to
+ *        the sequence length, which in turn is used to divide the score of the sequence. Since the score is the log
+ *        likelihood of the sequence (i.e. negative), `length_penalty` > 0.0 promotes longer sequences, while
+ *        `length_penalty` < 0.0 encourages shorter sequences.
+ * @param num_return_sequences the number of sequences to return for grouped beam search decoding.
+ * @param no_repeat_ngram_size if set to int > 0, all ngrams of that size can only occur once.
+ * @param stop_criteria controls the stopping condition for grouped beam search. It accepts the following values: 
+ *        "early", where the generation stops as soon as there are `num_beams` complete candidates; "heuristic", where an 
+ *        heuristic is applied and the generation stops when is it very unlikely to find better candidates;
+ *        "never", where the beam search procedure only stops when there cannot be better candidates (canonical beam search algorithm).
+ * 
+ * Random sampling parameters:
+ * @param temperature the value used to modulate token probabilities for random sampling.
+ * @param top_p - if set to float < 1, only the smallest set of most probable tokens with probabilities that add up to top_p or higher are kept for generation.
+ * @param top_k the number of highest probability vocabulary tokens to keep for top-k-filtering.
+ * @param do_sample whether or not to use multinomial random sampling that add up to `top_p` or higher are kept.
+ * @param repetition_penalty the parameter for repetition penalty. 1.0 means no penalty.
+ */
+class OPENVINO_GENAI_EXPORTS GenerationConfig {
+public:
+    GenerationConfig() = default;
+    explicit GenerationConfig(std::string json_path);
+
+    // Generic
+    size_t max_new_tokens = SIZE_MAX;
+    size_t max_length = SIZE_MAX;
+    bool ignore_eos = false;
+
+    // Beam search specific
+    size_t num_beam_groups = 1;
+    size_t num_beams = 1;
+    float diversity_penalty = 1.0f;
+    float length_penalty = 1.0f;
+    size_t num_return_sequences = 1;
+    size_t no_repeat_ngram_size = std::numeric_limits<size_t>::max();
+    StopCriteria stop_criteria = StopCriteria::heuristic;
+    
+    // Multinomial
+    float temperature = 0.0f;
+    float top_p = 1.0f;
+    int top_k = -1;
+    bool do_sample = false;
+    float repetition_penalty = 1.0f;
+
+    // special tokens
+    int64_t pad_token_id = 0;
+    int64_t bos_token_id = 1;
+    int64_t eos_token_id = 2;
+    
+    // used for chat scenario
+    std::string bos_token = "<s>";
+    std::string eos_token = "</s>";
+
+    size_t get_max_new_tokens(size_t prompt_length = 0) const;
+    bool is_greedy_decoding() const;
+    bool is_beam_search() const;
+    bool is_multimomial() const;
+    static GenerationConfig anymap_to_generation_config(const ov::AnyMap& config_map = {});
```

Comment:
Remove `static` and rename to `update_from_anymap()`

Reply:
done